src/TRE-API/Controllers/VaultCredentialsController.cs

```diff
@@ -0,0 +1,229 @@
+﻿using BL.Services.Contract;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Serilog;
+using Swashbuckle.AspNetCore.Annotations;
+using System.ComponentModel.DataAnnotations;
+
+namespace TRE_API.Controllers
+{
+    [Route("api/v1/vaultvredentials")]
+    [ApiController]
+    [SwaggerTag("Vayktcredentials", "Manage Credentials")]
+    [Produces("application/json")]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public class VaultCredentialsController : ControllerBase
+    {
+        private readonly IVaultCredentialsService _vaultService;
+        public VaultCredentialsController(IVaultCredentialsService vaultService)
+        {
+            _vaultService = vaultService ?? throw new ArgumentNullException(nameof(vaultService));
+        }
+
+        /// <summary>
+        /// Creates ephemeral credentials for a user's container execution
+        /// </summary>
+        /// <param name="request">The ephemeral credential request</param>
+        /// <returns>Success status and credential path</returns>
+        [HttpPost("ephemeral-credentials")]
```

Comment:
this will produce a route `/api/v1/vaultcredentials/ephemeral-credentials/`

Not sure if that was intended?

- Are we expecting `vaultcredentials` to have other non `ephemeral-credentials` routes later?
- Consistency of casing - if we kebab-case `ephemeral-credentials` we should also kebab-case `vault-credentials`? or kebab-case nothing. Doesn't matter which way but we should be consistent. I think some of the TRE credentials endpoints have merged with `tre-credentials` now